### PR TITLE
Bug 1173270 - Enable screen timeouts while using camera if in power save mode.

### PR DESCRIPTION
**Pull request by aosmond**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2b7af7a3035794f788470a2a61c91b43381d9314

File: apps/camera/js/controllers/battery.js [hunk 1]
```diff
@@ -41,6 +44,84 @@ BatteryController.prototype.bindEvents = function() {
...
+
+  var mozSettings = navigator.mozSettings;
+  mozSettings.addObserver('powersave.enabled', this.onPowerSaveChange);
+  mozSettings.createLock().get('powersave.enabled').then(
+    this.onPowerSaveChange);
+};
+
+BatteryController.prototype.onPowerSaveChange = function(values) {
+  var value;
+  if (typeof(values.settingValue) !== 'undefined') {
```
Comment (justindarc) [inline apps/camera/js/controllers/battery.js]:
```
if (values.settingValue !== undefined) {
```

File: apps/camera/js/controllers/battery.js [hunk 2]
```diff
@@ -41,6 +44,84 @@ BatteryController.prototype.bindEvents = function() {
...
+
+/**
+ * Restore the screen timeout to the original value before
+ * the camera was opened and power save mode was entered.
+ *
+ * @private
+ */
+BatteryController.prototype.restoreScreenTimeout = function() {
+  this.cachingScreenTimeout = false;
+  if (typeof(this.cachedScreenTimeout) === 'undefined') { return; }
```
Comment (justindarc) [inline apps/camera/js/controllers/battery.js]:
```
if (this.cachedScreenTimeout === undefined) {
```

File: apps/camera/js/controllers/battery.js [hunk 3]
```diff
@@ -41,6 +44,84 @@ BatteryController.prototype.bindEvents = function() {
...
+/**
+ * If in power save mode, change the screen timeout to the
+ * app-specific screen timeout, otherwise restore it to the
+ * original timeout period.
+ *
+ * @private
+ */
+BatteryController.prototype.updateScreenTimeout = function() {
+  if (this.app.hidden ||
+      typeof(this.powerSaveScreenTimeout) === 'undefined') {
```
Comment (justindarc) [inline apps/camera/js/controllers/battery.js]:
```
if (this.app.hidden || this. powerSaveScreenTimeout === undefined) {
```

File: apps/camera/js/controllers/battery.js [hunk 4]
```diff
@@ -41,6 +44,84 @@ BatteryController.prototype.bindEvents = function() {
...
+    return;
+  }
+
+  if (this.powerSave) {
+    if (typeof(this.cachedScreenTimeout) !== 'undefined' ||
+        this.cachingScreenTimeout) { return; }
```
Comment (justindarc) [inline apps/camera/js/controllers/battery.js]: if (this.cachedScreenTimeout !== undefined || this.cachingScreenTimeout) { return; }

**Pull request closed without merging**